manual/manual/refman/exten.etex

```diff
@@ -1315,6 +1315,53 @@ to implement dynamic types.
           | None -> None
           | Some Eq -> Some x
 \end{verbatim}
+\paragraph{Type variable names in error messages}
+The typing of pattern matching in presence of GADT can generate many
+existential types. When reporting a typing error, some of these
+existential types can be exposed and therefore need a
+compiler-generated name. The compiler names these variables accordingly
+to the following nomenclature:
+\begin{itemize}
+\item First, types whose name starts with a "$" are existentials.
+\item "$Constr_'a" denotes an existential type introduced for the type
+variable "'a" of the GADT constructor "Constr":
+\caml
+\?   type any = Any : \textquotesingle\-name -> any
+\?   let escape (Any x) = \<x\>;;
+\: Error: This expression has type $Any_\textquotesingle\-name
+\: but an expression was expected of type \textquotesingle\-a
+\: The type constructor $Any_\textquotesingle\-name would escape its scope
+\endcaml
+\item "$Constr" denotes an existential type introduced for an anonymous
+type variable in the GADT constructor "Constr":
+\caml
+\? type any = Any : _ -> any
+\? let escape (Any x) = \<x\>;;
+\: Error: This expression has type $Any but an expression was expected of type
+\:        \textquotesingle\-a
+\:        The type constructor $Any would escape its scope
+\endcaml
+\item "$'a" is an existential generated by reification of type
+variable "'a" during unification:
```

Comment:
I found this explanation confusing. I would rather say something like: "an existential variable may also be named `$'foo` if it was unified with the type variable `'foo`".